### PR TITLE
Add rtkheading2magheading configuration

### PR DESCRIPTION
**Pull request by wvmcastro**
[sc-59323]

Create rtkHeading2magHeading user configuration. When using gnss heading as source for orientation (instead of mags) this value will be added to the gnss heading value before feeding it to the filter.

### CODE REVIEW

File: lib/Core/Algorithm/src/algorithm.c [hunk 1]
```diff
@@ -125,6 +125,8 @@ void InitializeAlgorithmStruct(uint8_t callingFreq)
     gAlgorithm.pointOfInterestB[Y_AXIS] = 0.0;
     gAlgorithm.pointOfInterestB[Z_AXIS] = 0.0;
 
+    gAlgorithm.rtkHeading2magHeading = NAN;
```
Comment (doudou) [inline lib/Core/Algorithm/src/algorithm.c:128]: I think that zero would be a better default in this case. NAN has the option of blowing the filter completely.

File: src/user/UserConfiguration.c [hunk 2]
```diff
@@ -61,6 +61,7 @@ const UserConfigurationStruct gDefaultUserConfig = {
     .pointOfInterestBx   = 0.0,
     .pointOfInterestBy   = 0.0,
     .pointOfInterestBz   = 0.0,
+    .rtkHeading2magHeading = NAN,
```
Comment (doudou) [inline src/user/UserConfiguration.c:64]: Same here about the default.

File: lib/Core/Algorithm/src/algorithm.c (hunk 1 above)
Comment (wvmcastro) [reply lib/Core/Algorithm/src/algorithm.c:128]: I was planning to use it as a flag for "do not use rtk heading". That way we could enable/disable it

Comment (doudou) [reply lib/Core/Algorithm/src/algorithm.c:128]: Alrigth.

**Review by doudou — APPROVED** [review]

**Review by jhonasiv — APPROVED** [review]

**Pull request merged**